### PR TITLE
Trim bearer prefix no matter the case

### PR DESCRIPTION
**Pull request by pankrator**
## Motivation
if Authorization header is passed and bearer token is used, it should not matter in what case is the key word "bearer" 

## Pull Request status

* [x] Initial implementation

### CODE REVIEW

**Comment by dpanayotov** [comment]
Maybe a test that validates that the case doesn't matter

File: api/filters/authn/oauth/authenticator_test.go [hunk 1]
```diff
@@ -313,90 +311,135 @@ var _ = Describe("OIDC Authenticator", func() {
...
 
-				BeforeEach(func() {
-					verifier = &securityfakes.FakeTokenVerifier{}
-					authenticator = &oauthAuthenticator{Verifier: verifier}
-
-					request.Header.Set("Authorization", "Bearer token")
-				})
-
-				Context("when verifier returns an error", func() {
+				Context("when Bearer is starts with uppercase", func() {
```
Comment (KirilKabakchiev) [inline api/filters/authn/oauth/authenticator_test.go]: remove is

File: api/filters/authn/oauth/authenticator_test.go [hunk 2]
```diff
@@ -313,90 +311,135 @@ var _ = Describe("OIDC Authenticator", func() {
...
+						request.Header.Set("Authorization", "bearer token")
+					})
 
-						claims := struct {
-							Abc string
-						}{}
-						err = user.Claims(&claims)
-						Expect(err).To(BeNil())
-						Expect(claims.Abc).To(Equal("xyz"))
+					Context("when returned token is valid", func() {
```
Comment (KirilKabakchiev) [inline api/filters/authn/oauth/authenticator_test.go]: this seems pretty similar (duplicates) the test above. I don't insist but if you feel like, you could parameterize the tests (so you could cover all 3 scenarios for both lowercase and uppercase and avoid the code dup

File: api/filters/authn/oauth/authenticator.go [hunk 3]
```diff
@@ -96,13 +96,13 @@ func newOIDCConfig(options *options) *goidc.Config {
...
 	authorizationHeader := request.Header.Get("Authorization")
-	if authorizationHeader == "" || !strings.HasPrefix(strings.ToLower(authorizationHeader), "bearer") {
+	if authorizationHeader == "" || !strings.HasPrefix(strings.ToLower(authorizationHeader), "bearer ") {
 		return nil, security.Abstain, nil
 	}
 	if a.Verifier == nil {
 		return nil, security.Abstain, errors.New("authenticator is not configured")
 	}
-	token := strings.TrimPrefix(authorizationHeader, "Bearer ")
+	token := strings.TrimSpace(authorizationHeader[len("Bearer "):])
```
Comment (KirilKabakchiev) [inline api/filters/authn/oauth/authenticator.go]: not sure about this one, but as far as i remember the authentication token value (oauth2 spec) should be ignorecased. Meaning both Bearer and bearer should be expected here.

Comment (pankrator) [reply api/filters/authn/oauth/authenticator.go]: It doesn't matter what the case is. I use the length of the "Bearer " string.

File: api/filters/authn/oauth/authenticator_test.go (hunk 2 above)
Comment (pankrator) [reply api/filters/authn/oauth/authenticator_test.go]: Good catch, I forgot that I rearranged the tests.

File: api/filters/authn/oauth/authenticator.go (hunk 3 above)
Comment (KirilKabakchiev) [reply api/filters/authn/oauth/authenticator.go]: Yeah with this change you added len, my bad - for some reason i was looking at the trimpredix version

**Pull request merged**